### DISCUSSION TITLE
problem in writing tests formDSCEngine.sol

### DISCUSSION THREAD

**Discussion by khushbusandhu** [Cyfrin/foundry-full-course-cu#698] [General]
```

    modifier depositedCollateral() {
        vm.startPrank(USER);
        ERC20Mock(weth).approve(address(dsce), AMOUNT_COLLATERAL);
        dsce.depositeCollateral(weth, AMOUNT_COLLATERAL);
        _;
    }
```
 in modifierwe deposited collateral in dsce contract but in the test we are calling balanceof function on dsc contract
why so?
```
 function testCanDepositeCollateralWithoutMinting() public depositedCollateral {
        uint256 userBalance = dsc.balanceOf(USER);
        assertEq(userBalance, 0);
    }
```
 and how the balanceof dsc is equal to 0?

**Comment by shikhar229169** [comment]
The DSC contract inherits from ERC20Burnable contract which has the function balanceOf and it returns the amount of DSC the USER has. As the USER only deposited the collateral but no DSC was minted so the DSC balance should be 0.

**Reply by khushbusandhu** [reply]
okay got it!
one more question should I just copy paste the tests from the repo of the course ?

**Reply by shikhar229169** [reply]
No, never.
Write them on your own.

**Reply by khushbusandhu** [reply]
I have tried but I am not strong on all the concepts yet and I am facing a lot of problems in the same.
